### DISCUSSION TITLE
Connect to PostgreSQL over Unix socket

### DISCUSSION THREAD

**Discussion by gsiz007** [pgagroal/pgagroal#183] [General]
Actually is possible use in Server section parameters **host** and **port** both are mandatory.
Will be possible in future user unix socket in parameter **host** without mandatory parameter **port** ?

**Comment by jesperpedersen** [comment]
What is your main use-case for this ? Are you planning to deploy the entire stack on a single machine ?

Normally, you would want the connection pool on a separate machine than the PostgreSQL instance(s) in order to support scenarios like fail-over.

You can connect a client through Unix Domain Socket to a pgagroal instance to provide a local connection cache.

**Reply by gsiz007** [reply]
Typical use-case is heavy used REST API base on OpenResty on HTTPS conected together with pooler and PostgreSQL over unix socket.

**Comment by hunleyd** [comment]
> Normally, you would want the connection pool on a separate machine than the PostgreSQL instance(s) in order to support scenarios like fail-over.

we have plenty of customers that deploy a pooler on the pg nodes and use haproxy+patroni to route people to the current leader. 

we also have customers that have a pooler 'out in front' and then still have a poole on each pg node

maybe its not 'normal' but its certainly valid and something that should be supported

**Reply by jesperpedersen** [reply]
Should a leading `/` in `host` be the key to switch UDS ?

All the code should be there, so it is just a matter of deciding on the `pgagroal.conf` format for it.

**Comment by jesperpedersen** [comment]
I added support as above -- see #184.

You can test it with `master` or `1.3.x` -- alternative, use `localhost` for a TCP connection.

Thanks !

**Reply by gsiz007** [reply]
Thank you very much, I will check it.

**Comment by sheshjee** [comment]
pgagroal[3999010]: (libev) epoll_wait: Bad file descriptor
systemd[1]: Started Process Core Dump (PID 3999011/UID 0).
systemd-coredump[3999018]: Resource limits disable core dumping for process 3999010 (pgagroal).
systemd-coredump[3999018]: Process 3999010 (pgagroal) of user 26 dumped core.
systemd[1]: systemd-coredump@10-3999011-0.service: Succeeded.

I am getting this error multiple times when the client connects to my database. In thousand of connection request , this error occurs randomly.
